python.d/httpcheck.chart.py

```diff
@@ -0,0 +1,139 @@
+HTTP_NO_CONNECTION = 'no_connection'
+
+ORDER = ['response_time', 'response_length', 'status']
+
+CHARTS = {
+    'response_time': {
+        'options': [None, 'HTTP response time', 'ms', 'response', 'httpcheck.responsetime', 'line'],
+        'lines': [
+            [HTTP_RESPONSE_TIME, 'time', 'absolute', 100, 1000]
+        ]},
+    'response_length': {
+        'options': [None, 'HTTP response body length', 'characters', 'response', 'httpcheck.responselength', 'line'],
+        'lines': [
+            [HTTP_RESPONSE_LENGTH, 'length', 'absolute']
+        ]},
+    'status': {
+        'options': [None, 'HTTP status', 'boolean', 'status', 'httpcheck.status', 'line'],
+        'lines': [
+            [HTTP_SUCCESS, 'success', 'absolute'],
+            [HTTP_BAD_CONTENT, 'bad content', 'absolute'],
+            [HTTP_BAD_STATUS, 'bad status', 'absolute'],
+            [HTTP_TIMEOUT, 'timeout', 'absolute'],
+            [HTTP_NO_CONNECTION, 'no connection', 'absolute']
+        ]}
+}
+
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.url = self.configuration.get('url', None)
+        self.regex = re.compile(self.configuration.get('regex', '.*'))
+        self.status_codes_accepted = self.configuration.get('status_accepted', [200])
+        self.follow_redirect = self.configuration.get('redirect', True)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self._manager = None
+
+    def check(self):
+        """
+        Format configuration data and try to connect to server
+        :return: boolean
+        """
+        if not (self.url and isinstance(self.url, str)):
+            self.error('URL is not defined or type is not <str>')
+            return False
+
+        self._manager = self._build_manager()
+        if not self._manager:
+            return False
+
+        self.info('Enabled {url} with (redirect={redirect}, status={accepted}, interval={update}, timeout={timeout}, '
+                  'regex={regex})'
+                  .format(url=self.url, redirect=self.follow_redirect, accepted=self.status_codes_accepted,
+                          update=self.update_every, timeout=self.request_timeout, regex=self.regex.pattern))
+        return True
+
+    def _get_data(self):
+        """
+        Format data received from http request
+        :return: dict
+        """
+        data = dict()
+        data[HTTP_SUCCESS] = 0
+        data[HTTP_BAD_CONTENT] = 0
+        data[HTTP_BAD_STATUS] = 0
+        data[HTTP_TIMEOUT] = 0
+        data[HTTP_NO_CONNECTION] = 0
+        url = self.url
+        try:
+            retr = 1 if self.follow_redirect else False
+            start = time.time()
+            response = self._manager.request(
+                method='GET', url=url, timeout=self.request_timeout, retries=retr, headers=self._manager.headers,
+                redirect=self.follow_redirect
```

Comment:
Well, if you need i suggest to extend `UrlService._get_raw_data()` method instead of doing this.

Reply:
The only thing missing in current implementation is `redirect`

Reply:
And yes, we can return status code and content from `UrlService._get_raw_data()` ofc

Reply:
What about catching the various kinds of exceptions? HTTPerror is too generic for httpcheck...

Reply:
Well, i see no exception handling in `_get_raw_data_with_status()` in my prev message.
You have to catch them in your module.